libutils/logging.c

```diff
@@ -277,25 +277,33 @@ void VLog(LogLevel level, const char *fmt, va_list ap)
     free(hooked_msg);
 }
 
+/* TODO create libutils/defs.h. */
+#define CF_MAX_BUFSIZE 16384
+
 /**
- * @brief Logs binary data in #buf, with each byte translated to '.' if not
- *        printable. Message is prefixed with #prefix.
+ * @brief Logs binary data in #buf, with unprintable bytes translated to '.'.
+ *        Message is prefixed with #prefix.
+ * @param #buflen must be no more than CF_MAX_BUFSIZE
  */
 void LogRaw(LogLevel level, const char *prefix, const void *buf, size_t buflen)
 {
-    /* Translate non printable characters to printable ones. */
-    const unsigned char *src = (const unsigned char *) buf;
+    const unsigned char *src = buf;
     unsigned char dst[buflen+1];
-    size_t i;
+    assert(sizeof(dst) <= CF_MAX_BUFSIZE);
```

Comment:
I totally agree with truncating extra long messages, but `LogRaw()` is not
called from `Log()`, but it calls `Log()` itself. For now it's used to
print network buffers, which are guaranteed to be less than CF_BUFSIZE.

On Mon, Jan 12, 2015 at 11:00 AM, Kristian Amlie <notifications@github.com>
wrote:

> In libutils/logging.c
> <https://github.com/cfengine/core/pull/2086#discussion-diff-22781661>:
>
> >      unsigned char dst[buflen+1];
> > -    size_t i;
> > +    assert(sizeof(dst) <= CF_MAX_BUFSIZE);
>
> LogRaw is anything that comes from Log, isn't it? Since this is
> potentially dependent on user data (if you have some long JSON string or
> something), I think we shouldn't assert on overflowing the length.
>
> I do agree that allocating on the stack with no check at all is bad
> though. Maybe truncate message instead, like What does the fox sa....
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfengine/core/pull/2086/files#r22781661>.
>